File: [0055] Jump Game - Greedy.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        greatest_can_reach=0

        for ii, vv in enumerate(nums):
            if ii > greatest_can_reach: return False

            greatest_can_reach=max(greatest_can_reach, ii+nums[ii])


        return True if greatest_can_reach >= len(nums)-1 else False
        

File: [0055] Jump Game - Greedy.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {

        int far = 0;
        for(int i = 0; i<nums.size()-1; i++) //simulating moving idx by idx
        {
            if (far < i) return false;//we cannot move beyound the farest we could
            far = max(far,i+nums[i]);
        }

        return far>=nums.size()-1;
    }
};
